### ISSUE TITLE
Test placeholder transformations

### ISSUE THREAD

**Issue by jrieken** [Microsoft/vscode#52796]
Refs: #34683 

- [ ] any
- [ ] any

Complexity: 3

Snippet transformation allow you to modify the value of a placeholder before moving on to the next placeholder. Refer to the documentation about all details: https://github.com/Microsoft/vscode/blob/master/src/vs/editor/contrib/snippet/browser/snippet.md#L31. In short, transforms are four values: (1) the placeholder to transform, (2) the regex that describes the transform, (3) a replace string that is inserted, (4) regex options. Also note that the syntax is "given" by Textmate, so just make sure we support what we have promised.

Test that

* you can upper-case (lower-case, capitalise) things, e.g with `$1 becomes: ${1/(.*)/${1:/upcase}/g}`
* you can generate a getter/setting from a property 
* get creative, check the doc linked above is good

**Comment by ralexand56** [comment]
I don't understand. Does this code work? $1 becomes: ${1/(.*)/${1:/upcase}/g}

Can I actually replace with uppercase characters now?

**Comment by chrmarti** [comment]
@ralexand56 Correct, there is some documentation here: https://github.com/Microsoft/vscode/blob/master/src/vs/editor/contrib/snippet/snippet.md#placeholder-transform
